### DISCUSSION TITLE
Howto insert captcha?

### DISCUSSION THREAD

**Discussion by sgbossinga** [bimmerconnected/bimmer_connected#703] [Q&A] [answered]
Hi, I installed the new HA versions to re-enable the BMW integration but how do I insert the captcha? 
Where can I issue the command: bimmerconnected status USERNAME PASSWORD REGION --captcha-token CAPTCHA_TOKEN?
How can I get the CLI started?

I am not a developer and need a little more than what the documentation gives me.

**Comment by rikroe** [comment]
Ou don't need to worry about this. Just go to your HA Devices & Services setting and it will ask you to reauthenticate/reconfigure.
You will be asked for the captcha during this process.

**Reply by sgbossinga** [reply]
sorry but it doesn't

**Reply by rikroe** [reply]
What doesn't? If you have upgraded HA and disabled the integration before, you need to reactivate it

**Reply by sgbossinga** [reply]
I upgraded both OS and the other 'thing' today but didn't disable the integration

**Reply by sgbossinga** [reply]
still get the invalid_auth message

**Reply by rikroe** [reply]
Have you upgraded in the last 2 hours? OS version doesn't matter, it is the actual HA programm running on it.
Settings > About for checking

**Reply by sgbossinga** [reply]
Core
2024.12.0
Supervisor
2024.11.4
Operating System
14.0
Frontend
20241106.2

**Reply by rikroe** [reply]
This should work from a software side.

Please check if you are able to log in to the MyBMW website. Most people today had issues with their login data being incorrect.

**Reply by sgbossinga** [reply]
it works fine

**Reply by sgbossinga** [reply]
should the captcha screen follow the username/password/region popup?

**Reply by rikroe** [reply]
Yes

**Reply by sgbossinga** [reply]
well that doesn't work (I'm in The Netherlands/EU)

**Reply by sgbossinga** [reply]
2024-12-04 23:36:23.502 DEBUG (MainThread) [bimmer_connected.api.authentication] Authenticating with MyBMW flow for North America & Rest of World.
2024-12-04 23:36:23.773 ERROR (MainThread) [bimmer_connected.api.authentication] MyBMWAuthError due to HTTPStatusError: invalid_client - Client authentication failed (e.g., login failure, unknown client, no client authentication included or unsupported authentication method)

**Reply by sgbossinga** [reply]
2024-12-04 23:35:35.480 WARNING (MainThread) [homeassistant.config_entries] Config entry '**my email' for bmw_connected_drive integration could not authenticate: HTTPStatusError: invalid_client - Client authentication failed (e.g., login failure, unknown client, no client authentication included or unsupported authentication method)

**Reply by rikroe** [reply]
It seems something hasn't fully updated or so. Can you try tomorrow after a few restart?

**Reply by sgbossinga** [reply]
sure

**Comment by sgbossinga** [comment]
Tried again after multiple restarts and server reboots, no captcha popup after usernane/password/region

**Reply by sgbossinga** [reply, marked as answer]
ok found it, the custom component was not updated. After that, I entered the captcha in the popup and the integration is working again

**Reply by rikroe** [reply]
Good 👍

Please remove the custom component completely as it won't get updates in the future.

**Reply by sgbossinga** [reply]
> Good 👍
> 
> Please remove the custom component completely as it won't get updates in the future.

but then I loose the device (car) if I disable BMW Connected Drive BETA

**Reply by rikroe** [reply]
Not disable. Remove completely from HACS, after HA restart it will use the version integrated with HA.

**Reply by rikroe** [reply]
To be clear do this (assume this would be the BETA):

![image](https://github.com/user-attachments/assets/6e966427-6fe4-452f-9332-7e2e9463216d)

**Reply by sgbossinga** [reply]
Yes thanks @rikroe  after the reboot no more beta

and thanks for all your great work
